hedera-node/src/main/java/com/hedera/services/store/TopicStore.java

```diff
@@ -62,4 +70,42 @@ public void persistNew(Topic model) {
 		currentTopics.put(id, merkleTopic);
 		transactionRecordService.includeChangesToTopic(model);
 	}
+
+	/**
+	 * Returns a model of the requested topic, with operations that can be used to
+	 * implement business logic in a transaction.
+	 *
+	 * <b>IMPORTANT:</b> Changes to the returned model are not automatically persisted
+	 * to state! The altered model must be passed to {@link TopicStore#persistTopic(Topic)}
+	 * in order for its changes to be applied to the Swirlds state, and included in the
+	 * {@link com.hedera.services.state.submerkle.ExpirableTxnRecord} for the active transaction.
+	 *
+	 * @param id
+	 * 		the topic to load
+	 * @return a usable model of the topic
+	 * @throws InvalidTransactionException
+	 * 		if the requested topic is missing, deleted, or expired and pending removal
+	 */
+	public Topic loadTopic(Id id) {
+		final var merkleTopic = topics.get().get(EntityNum.fromLong(id.getNum()));
+		validateUsable(merkleTopic);
+		return fromMerkle(merkleTopic, id);
+	}
+
+	/**
+	 * Takes a Topic model and parses it to the corresponding MerkleTopic.
+	 *
+	 * @param topic
+	 * 		the topic model
+	 */
+	public void persistTopic(Topic topic) {
+		final var key = EntityNum.fromLong(topic.getId().getNum());
+		var mutableTopic = topics.get().getForModify(key);
+		modelToMerkle(topic, mutableTopic);
+	}
+
+	private void validateUsable(MerkleTopic merkleTopic) {
+		validateFalse(merkleTopic == null || merkleTopic.isDeleted(), INVALID_TOPIC_ID);
+		validateTrue(merkleTopic.hasAdminKey(), UNAUTHORIZED);
```

Comment:
I think this may behave differently when merged with `SubmitMessage` - the admin key check

Reply:
I think you can now safely remove row 114